### PR TITLE
[BD-27][TNL-7891][BB-3539] Remove Empty Sections from cc2olx Conversions - ContextModuleSubHeader

### PR DESCRIPTION
**Pull request by shimulch**
This PR fixes an issue related to empty sections while there are ``canvas`` items with content type ``ContextModuleSubHeader``. With this PR, ``cc2olx`` will treat these types of items as ``subsections`` and collapse related (consecutive) items under it as ``verticals``.

**JIRA tickets**: 
- https://tasks.opencraft.com/browse/BB-3539
- https://openedx.atlassian.net/browse/TNL-7891

~~**Discussions**:~~

**Dependencies**: None

~~**Screenshots**:~~

~~**Sandbox URL**:~~

**Merge deadline**: None

**Testing instructions**:

1. Pull this PR.
2. Find a canvas ``imscc`` file that contains ``ContextModuleSubHeader``, you can use the updated fixture with this PR.
3. Convert using ``cc2olx``. 
4. Check that item after ``ContextModuleSubHeader`` collapsed into it as verticals.

**Author notes and concerns**:
N/A

**Reviewers**
- [ ] @kaizoku
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shimulch! I've created [OSPR-5615](https://openedx.atlassian.net/browse/OSPR-5615) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: src/cc2olx/models.py [hunk 1]
```diff
@@ -123,7 +171,7 @@ def normalize(self):
             course_root = course_root[0]
         if not course_root:
             return
-        sections = course_root.get("children", [])
+        sections = self.process_canvas_cc(course_root.get("children", []))
```
Comment (kaizoku) [inline src/cc2olx/models.py:174]: Should we run this in all cases, or only run this transformation for canvas flavored common cartridges?
We can check if `course_settings/canvas_export.txt` exists to determine if this is a canvas flavored CC.
Canvas [writes this file to determine if it's from a canvas export](https://github.com/instructure/canvas-lms/blob/8a3f0249376ca246106f6505063e68b5829fcfac/lib/cc/canvas_resource.rb#L90).

Comment (shimulch) [reply src/cc2olx/models.py]: @kaizoku, Thanks for the suggestion. Yeah this approach would be better. I've updated my PR.

**Comment by kaizoku** [comment]
Thanks for the PR @shimulch.
This works well for me on the affected courses and isn't causing issues with other courses. :+1:

- [✓] I tested this: (converted courses affected and not affected by issues, examined outlines and course structure in studio/lms)
- [✓] I read through the code
- [n/a] ~I checked for accessibility issues~
- [n/a] ~Includes documentation~

**Review by kaizoku — APPROVED** [review]

**Review by alangsto — APPROVED** [review]

**Pull request merged**